qa-pipelines/tasks/lib/prepare-kubeconfig.sh

```diff
@@ -5,10 +5,44 @@ read_yaml_key() {
     ruby -r yaml -e "puts YAML.load_file('$1')[\"$2\"]"
 }
 
+aws_auth_cm_yaml() {
```

Comment:
Can you add some comments about what this does? And possibly move into a separate file

Reply:
it is part of comments. From line 37 to 41

Reply:
that's not really a comment, and it doesn't explain what eksServiceRole is or why we need it.

Reply:
The IAM user that created the cluster is hardwired in the AWS auth config map. So this is updating the config map to point to a different user/role e.g. eksServiceRole to avoid the non-portability of kubeconfig.

https://docs.aws.amazon.com/eks/latest/userguide/create-kubeconfig.html